### PR TITLE
[dev-v2.6-0123-patches] Add preserveUnknownFields: false in calico and canal template

### PR DESCRIPTION
**Pull request by rayandas**
Fixes the issue: https://github.com/rancher/rancher/issues/40280  and https://github.com/rancher/rke/issues/3152 (The changes are tested on Rancher v2.6.9)
This changes will allow updating from the v1beta1 to the v1 API for  felixconfiguration and ipamblock CRDs.
This PR also adds a new template `v1.22.17-rancher1-2` so that we can align it with `calico_v3.22-rancher2` and `canal_v3.22-rancher2` in templates.go

Steps to create the bug:
1. Setup Rancher `v2.6.5` using docker
2. Create a `v1.18.20-rancher1` cluster.
3. Upgrade to Rancher `v2.6.9` using these steps
4. Upgrade the existing `v1.18.20-rancher1`  cluster to `v1.24.9-rancher1` 

You'll see the job `rke-network-plugin-deploy-job` is failing.

- [x] TODO: Forward-port it to `dev-v2.7`

### CODE REVIEW

**Review by snasovich — DISMISSED** [review]
LGTM, confirmed the updated templates define `preserveUnknownFields: false` for a couple of CRDs mentioned on the issue and update template application version ranges look good too.

Good to merge once @kinarashah approves.

**Comment by kinarashah** [comment]
Two small changes from dev-v2.7 PR. Apologies for nitpicking, we've run into issues if the rancher changes aren't explicitly called out in the template when introducing a new template.

**Comment by rayandas** [comment]
@kinarashah Not a problem. Updated the PR with your suggestions.

**Review by snasovich — APPROVED** [review]

**Review by kinarashah — APPROVED** [review]

**Pull request merged**
